bin/fixmystreet.com/call-wasteworks-backend

```diff
@@ -37,10 +38,42 @@ my $cfg = $cobrand->feature($opts->backend);
 my $class = $opts->backend eq 'echo' ? 'Integrations::Echo' : 'Integrations::Bartec';
 my $integration = $class->new(%$cfg);
 
+my $lockfile;
+if ($opts->fork) {
+    unless ($opts->out) {
+        die "Must use --out with --fork";
+    }
+
+    $lockfile = $opts->out . ".lock";
```

Comment:
An alternative here would be (I think?) to open the $opts->out file first off (with fork or not), `flock` it exclusively, then use store_fd to save to the already open handle. The front end would then: if file doesn't exist, run this script with `&` to background it immediately; if it does exist, try and get a non-blocking shared lock on the output file - if that fails, it's still being looked up, show loading page; if that succeeds, load in the data. I don't think anything else in this script would then need to change - it wouldn't have to worry about anything at all really, it doesn't need to fork or think about things like that. Does that work? Will come back to this next week.

Reply:
Have implemented this logic in d1aeec07589365399b0f2e323bfc1f1728c00537. Works well locally, will see how it goes on staging.